programs/ziti-edge-tunnel/instance.c

```diff
@@ -354,7 +382,9 @@ tunnel_identity_array get_tunnel_identities() {
 
     int idx = 0;
     MODEL_MAP_FOREACH(id, tnl_id, &tnl_identity_map) {
-        tnl_id_arr[idx++] = tnl_id;
+        if (tnl_id->Status == instance_status_ok) {
```

Comment:
Why not move this test outside of the foreach, to avoid unnecessarily iterating the map?

    if (tnl_id->Status == instance_status_ok) {
        MODEL_MAP_FOREACH(id, tnl_id, &tnl_identity_map) {
            tnl_id_arr[idx++] = tnl_id;
        }
    }

Reply:
tnl_id is the value of each entry in the map, that is why I am checking this condition from inside the loop.